tests/functional/dispersionCoarse/dispersion1d.py

```diff
@@ -0,0 +1,379 @@
+            quantity=quantity,
+            write_timestamps=timestamps,
+            compute_timestamps=timestamps,
+        )
+
+
+    for quantity in ["density", "bulkVelocity"]:
+        FluidDiagnostics(
+            quantity=quantity,
+            write_timestamps=timestamps,
+            compute_timestamps=timestamps,
+            )
+
+    return wave_numbers, b_amplitudes
+
+
+
+# ___ post-processing functions
+def get_all_w(run_path, wave_numbers, polarization):
+    file = os.path.join(run_path, "EM_B.h5")
+    times = get_times_from_h5(file)
+
+    nm = len(wave_numbers)
+    print('number of modes : {}'.format(nm))
+
+    r = Run(run_path)
+    byz = np.array([])
+
+    for time in times:
+
+        interp_by, x = r.GetB(time, merged=True, interp='nearest')['By']
+        interp_bz, x = r.GetB(time, merged=True, interp='nearest')['Bz']
+        by = interp_by(x[0])
+        bz = interp_bz(x[0])
+
+        # polarization = +1 for R mode, -1 for L mode
+        byz = np.concatenate((byz, by+polarization*1j*bz))
+
+    nx = x[0].shape[0]
+    nt = times.shape[0]
+    byz = np.reshape(byz, (nt, nx))
+
+    BYZ = np.absolute(np.fft.fft2(byz)[:(nt+1)//2, :(nx+1)//2])
+    BYZ_4_all_W = np.sum(BYZ, axis=0)
+
+    idx = np.argsort(BYZ_4_all_W)
+    kmodes = idx[-nm:]
+
+    wmodes = np.array([])
+    for i in range(nm):
+        wmodes = np.append(wmodes, np.argmax(BYZ[:,idx[-nm+i]]))
+        #wmodes.append(np.argmax(BYZ[:,idx[-nm+i]]))
+
+    idx = np.argsort(kmodes)
+
+    print(kmodes[idx], wmodes[idx])
+
+    return kmodes[idx], wmodes[idx], BYZ
+
+
+
+def main():
+    # list of modes : m = 1 is for 1 wavelength in the whole domain
+    modes = [4, 8, 16, 32, 64, 128, 256, 512]
+
+    # lists of amplitudes of the magnetic field amplitudes
+    b_amplitudes = [0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01]
+
+    # polarization : -1 for L mode
+    wave_nums, b1 = setOfModes(-1, modes, b_amplitudes, cpp.mpi_rank()+1)
+    simulator = Simulator(gv.sim)
+    simulator.initialize()
+    simulator.run()
+
+    from pybindlibs.cpp import mpi_rank
+    from matplotlib import rc
+
+    if mpi_rank() == 0:
+        sim = ph.global_vars.sim
+
+        L = sim.simulation_domain()[0]
+        T = sim.final_time
+
+        #for the left mode
+        ki, wi, byz = get_all_w(os.path.join(os.curdir, "setOfModes1d"), wave_nums, -1)
+
+        np.save('left1d.npy', byz)
+
+        k_numL = 2*np.pi*ki/L
+        w_numL = 2*np.pi*wi/T
+
+    ph.global_vars.sim = None
+
+    # list of modes : m = 1 is for 1 wavelength in the whole domain
+    modes = [4, 8, 16, 32, 64, 128, 256, 512]
+
+    # lists of amplitudes of the magnetic field amplitudes
+    b_amplitudes = [0.005, 0.005, 0.005, 0.005, 0.005, 0.005, 0.005, 0.005]
+
+    # polarization : +1 for R mode
+    wave_nums, b1 = setOfModes(+1, modes, b_amplitudes, cpp.mpi_rank()+1)
+    simulator = Simulator(gv.sim)
+    simulator.initialize()
+    simulator.run()
+
+    if mpi_rank() == 0:
```

Comment:
cpp.mpirank()

or you can reference the function via

```
mpi_rank = cpp.mpi_rank
```
somewhere near the top of this file